### PR TITLE
samples: enable bucket lifecycle management

### PR DESCRIPTION
**Pull request by jkwlui**

### CODE REVIEW

File: samples/system-test/bucketLifecycle.test.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+  after(async () => {
+    await bucket.delete().catch(console.error);
+  });
+
+  it('should add a lifecycle delete rule', async () => {
+    const output = execSync(
+      `node enableBucketLifecycleManagement.js ${bucketName}`
+    );
+    assert.match(
```
Comment (JustinBeckwith) [inline samples/system-test/bucketLifecycle.test.js:46]: Instead, may be easier to do:

```js
assert.include(output, `Lifecycle management is enable for bucket ${bucketName} and the rules are:`);
```

**Review by JustinBeckwith — APPROVED** [review]

File: samples/enableBucketLifecycleManagement.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  // Imports the Google Cloud client library
+  const {Storage} = require('@google-cloud/storage');
+
+  // Creates a client
+  const storage = new Storage();
+
+  async function enableBucketLifecycleManagement() {
+    const [metadata] = await storage.bucket(bucketName).addLifecycleRule({
+      action: 'delete',
+      condition: {age: 2},
```
Comment (frankyn) [inline samples/enableBucketLifecycleManagement.js:39]: Following canonical Java:

```java
// See the LifecycleRule documentation for additional info on what you can do with lifecycle
    // management rules. This one deletes objects that are over 100 days old.
    // https://googleapis.dev/java/google-cloud-clients/latest/com/google/cloud/storage/BucketInfo.LifecycleRule.html
    bucket
        .toBuilder()
        .setLifecycleRules(
            ImmutableList.of(
                new LifecycleRule(
                    LifecycleRule.LifecycleAction.newDeleteAction(),
                    LifecycleRule.LifecycleCondition.newBuilder().setAge(100).build())))
```

**Review by frankyn — CHANGES_REQUESTED** [review]

**Review by frankyn — APPROVED** [review]

File: samples/enableBucketLifecycleManagement.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  const storage = new Storage();
+
+  async function enableBucketLifecycleManagement() {
+    const [metadata] = await storage.bucket(bucketName).addLifecycleRule({
+      action: 'delete',
+      condition: {age: 100},
+    });
+
+    console.log(
+      `Lifecycle management is enable for bucket ${bucketName} and the rules are:`
```
Comment (stephenplusplus) [inline samples/enableBucketLifecycleManagement.js:43]:
```suggestion
      `Lifecycle management is enabled for bucket ${bucketName} and the rules are:`
```

File: samples/system-test/bucketLifecycle.test.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    await bucket.delete().catch(console.error);
+  });
+
+  it('should add a lifecycle delete rule', async () => {
+    const output = execSync(
+      `node enableBucketLifecycleManagement.js ${bucketName}`
+    );
+    assert.include(
+      output,
+      `Lifecycle management is enable for bucket ${bucketName} and the rules are:`
```
Comment (stephenplusplus) [inline samples/system-test/bucketLifecycle.test.js:48]:
```suggestion
      `Lifecycle management is enabled for bucket ${bucketName} and the rules are:`
```

File: samples/enableBucketLifecycleManagement.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * This application demonstrates how to enable Object Lifecycle Management for
+ * a bucket.
+ *
+ * For more information, see the README.md under /storage and the documentation
```
Comment (stephenplusplus) [inline samples/enableBucketLifecycleManagement.js:19]: I just noticed we say this other places, but what do we mean by under "/storage"? We have one in the root of the repo which would be under "nodejs-storage" if they cloned the repo, and one in samples/. Maybe we should just put the URL to the GitHub-hosted file that we mean?

**Review by stephenplusplus — CHANGES_REQUESTED** [review]

File: samples/enableBucketLifecycleManagement.js (hunk 5 above)
Comment (jkwlui) [reply samples/enableBucketLifecycleManagement.js]: Seems like a relic from the past. Rewrote to just point to lifecycle docs.

**Review by stephenplusplus — APPROVED** [review]

**Pull request merged**